src/particles/integrators/Integrators.H

```diff
@@ -0,0 +1,119 @@
+/* Copyright 2022 The Regents of the University of California, through Lawrence
+ *           Berkeley National Laboratory (subject to receipt of any required
+ *           approvals from the U.S. Dept. of Energy). All rights reserved.
+ *
+ * This file is part of ImpactX.
+ *
+ * Authors: Chad Mitchell, Axel Huebl
+ * License: BSD-3-Clause-LBNL
+ */
+#ifndef IMPACTX_INTEGRATORS_H_
+#define IMPACTX_INTEGRATORS_H_
+
+#include <ablastr/particles/IndexHandling.H>
+
+#include <AMReX_Extension.H>  // for AMREX_RESTRICT
+#include <AMReX_REAL.H>       // for ParticleReal
+#include <AMReX_Print.H>      // for PrintToFile
+
+
+namespace impactx::integrators
+{
+
+    template <typename T_Element>
+    void symp2_integrate (
+        RefPart & refpart,
+        amrex::ParticleReal const zin,
+        amrex::ParticleReal const zout,
+        int const nsteps,
+        T_Element const & element
+    )
+    {
+        using namespace amrex::literals; // for _rt and _prt
+
+        // initialize numerical integration parameters
+        amrex::ParticleReal const dz = (zout-zin)/nsteps;
+        amrex::ParticleReal const tau1 = dz/2.0_prt;
+        amrex::ParticleReal const tau2 = dz;
+
+        // initialize the value of the independent variable
+        amrex::ParticleReal zeval = zin;
+
+        // loop over integration steps
+        for(int j=0; j < nsteps; ++j)
+        {
+            element.map1(tau1,refpart,zeval);
+            element.map2(tau2,refpart,zeval);
+            element.map1(tau1,refpart,zeval);
+        }
+    }
+
+    template <typename T_Element>
```

Comment:
We need to add the "hostdevice" and "inline" macros here: 
```suggestion
    template <typename T_Element>
    AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
```